### DISCUSSION TITLE
Docs: Tweaks to "strings" documentation

### DISCUSSION THREAD

**Discussion by richb-hanover** [PRQL/prql#1144] [General]
Random thoughts for improving the discussion of strings:

- Re-order the topics for strings in the Book. Perhaps:
   4.4 - Strings
   4.5 - F-Strings
   4.6 - S-strings
- Main "Strings" page should document the use of `'`, `"`, `'''`, `"""`, and `' ... "..."...'`
- Main "Strings" page should document substitution of variables using braces - `{x}`
- I am having trouble describing the difference between "F-Strings" and "S-Strings". Their pages aren't exactly clear about the definition: the examples currently only provide vague "do it like this" hints.
   - F-strings have something to do with concatenating the values
   - S-strings just do variable `{x}` substitution (I think)

I'm not sure if this is a "Discussion" or just an "Issue" - I'm trying a Discussion to see how it feels.

**Comment by max-sixty** [comment]
Thanks @richb-hanover — let me respond here briefly.

If you'd be up for a PR with the changes, that would be a great contribution. I think that's by far the most effective way to turn this sort of feedback into something that compounds to make the project better — similar to my thoughts in https://github.com/prql/prql/issues/1124. Happy to help with any setup etc; running `task setup-dev` should get you everything you need.

>    Re-order the topics for strings in the Book. Perhaps:

The order is currently alphabetical, though we could have a "See Also" section at the bottom of each? Or name them: `Strings: S-strings` and retain the alphabetical order while grouping them?

>    Main "Strings" page should document substitution of variables using braces - `{x}`

The substitution is only in s or f-strings.

>    I am having trouble describing the difference between "F-Strings" and "S-Strings". Their pages aren't exactly clear about the definition: the examples currently only provide vague "do it like this" hints.

F-strings create a string column like `f"{first}_{last}"` -> `CONCAT(first, "_", last)`, and almost always involve `CONCAT`.

S-strings create actual SQL, like `s"version()"` -> `version()`.

**Reply by richb-hanover** [reply]
These are all helpful points. Somewhat OT thoughts - not sure where else I could say them: 

* Thanks for encouraging me to make these changes myself. I have to balance my energies between having fun with PRQL and $DAYJOB - part of which involves actually massaging those tables of property values. (This is my gentle way of saying I may not get round to making those changes. :-)
* I threw a note over the wall to the folks at [sqlitetutorial.net](sqlitetutorial.net) to see if they might incorporate PRQL into their [Try It](https://www.sqlitetutorial.net/tryit/) interface. "All they would need to do" is let me load a different database, and use the prql-compiler to generate the SQL send to the DB. No response after 12 hours...
* I am extremely impressed by the maturity of this project, even in these early days. The fact that you have the Playground, a VScode extension, PyPRQL, prql-query, and a build environment working in concert shows the project "has heart."
* I also appreciate your organization and leadership of this project. I have the sense that there are a small number of core developers, and a bunch of hangers-on whom you make feel extremely welcome. Your willingness to accept my thoughts and PRs is terrific. 

Thanks again.

**Reply by max-sixty** [reply]
Thanks a lot @richb-hanover !
